### PR TITLE
Fix CleanUpIT. Fixes #1449

### LINKED ISSUE #1449: CleanUpIT fails after #1374
I've been seeing this test failure since #1374 was merged into the master branch:

```java
java.lang.AssertionError: Threads did not go away. Saw 1
	at org.junit.Assert.fail(Assert.java:88)
	at org.apache.accumulo.test.functional.CleanUpIT.run(CleanUpIT.java:148)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:298)
	at org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:292)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.lang.Thread.run(Thread.java:834)
```

### PR DESCRIPTION
**Pull request by Manno15**
CleanUpIt was failing due to 'Thread pool checker' not being closed properly. With this change, the thread is picked up properly and closed as expected. 

#1449

### CODE REVIEW

File: core/src/main/java/org/apache/accumulo/core/clientImpl/ThriftTransportPool.java [hunk 1]
```diff
@@ -861,10 +849,18 @@ public void startCheckerThread() {
...
+
+    for (Entry<ThriftTransportKey,CachedConnections> entry : pool.connections.entrySet()) {
+      Iterator<CachedConnection> iter = entry.getValue().unreserved.iterator();
+      while (iter.hasNext()) {
+        pool.executeWithinLock(entry.getKey(), (key) -> {
+          entry.getValue().removeExpiredConnections(expiredConnections, killTime);
+          entry.getValue().checkReservedForStuckIO();
+        });
+      }
+    }
```
Comment (ctubbsii) [inline core/src/main/java/org/apache/accumulo/core/clientImpl/ThriftTransportPool.java]: I don't understand why this is iterating over the `pool.connections.entrySet()`, but is checking for a non-empty iterator from `entry.getValue().unreserved`. Is the `iter.hasNext()` even necessary? The iterator doesn't seem to be advanced by calling `next()` ever. Is that check performing something important?

Comment (Manno15) [reply core/src/main/java/org/apache/accumulo/core/clientImpl/ThriftTransportPool.java]: This was similar to how it was handled before the refactoring. Without the Iter, the thread never gets picked up and removed.

Comment (Manno15) [reply core/src/main/java/org/apache/accumulo/core/clientImpl/ThriftTransportPool.java]: There are a couple of other loop options that seem to work but none that are just the for loop.

Comment (keith-turner) [reply core/src/main/java/org/apache/accumulo/core/clientImpl/ThriftTransportPool.java]: I was having a hard time understanding this change.   I attempted to fix this issue in #1477

Comment (Manno15) [reply core/src/main/java/org/apache/accumulo/core/clientImpl/ThriftTransportPool.java]: I will take a look at your changes. My goal was to fix the bug without reverting too much of the refactor that was done to it. In an older release, a similar method was done.

Comment (Manno15) [reply core/src/main/java/org/apache/accumulo/core/clientImpl/ThriftTransportPool.java]: I will take a look at your changes. My goal was to fix the bug without reverting too much of the refactor that was done to it. In an older release, a similar method was done.

**Comment by ctubbsii** [comment]
Closing this. It was superseded by #1477.

**Pull request closed without merging**